lib/openhab/dsl/items/date_time_item.rb

```diff
@@ -53,6 +55,7 @@ def coerce(other)
         # @!visibility private
         def format_type(command)
           return command.iso8601 if command.respond_to?(:iso8601)
+          return DateTimeFormatter::ISO_OFFSET_DATE_TIME.format(command) if command.is_a? Java::JavaTime::ZonedDateTime
```

Comment:
I feel like we've standardized on java.time.ZonedDateTime format for referencing java classes where possible.